irispy/iris_tools.py

```diff
@@ -473,14 +395,351 @@ def _calculate_orbital_wavelength_variation(data_array, date_data_created, slit_
+    Returns
+    -------
+    new_data_arrays: `list` of `numpy.ndarray`s
+        Data arrays with exposure time corrected for.
+
+    new_unit_time_accounted: `astropy.unit.Unit`
+        Unit of new data arrays after exposure time correction.
+
+    """
+    if u.s not in old_unit.decompose().bases:
+        new_data_arrays = [old_data/exposure_time for old_data in old_data_arrays]
+        new_unit = old_unit/u.s
+    else:
+        new_data_arrays = old_data_arrays
+        new_unit = old_unit
+    return new_data_arrays, new_unit
+
+def uncalculate_exposure_time_correction(old_data_arrays, old_unit, exposure_time):
+    """
+    Removes exposure time correction from data arrays.
+
+    Parameters
+    ----------
+    old_data_arrays: iterable of `numpy.ndarray`s
+        Arrays of data to be converted.
+
+    old_unit: `astropy.unit.Unit`
+        Unit of data arrays.
+
+    exposure_time: `numpy.ndarray`
+        Exposure time in seconds for each exposure in data arrays.
+
+    Returns
+    -------
+    new_data_arrays: `list` of `numpy.ndarray`s
+        Data arrays with exposure time correction removed.
+
+    new_unit_time_accounted: `astropy.unit.Unit`
+        Unit of new data arrays after exposure time correction removed.
+
+    """
+    if u.s not in (old_unit*u.s).decompose().bases:
+        new_data_arrays = [old_data * exposure_time for old_data in old_data_arrays]
+        new_unit = old_unit*u.s
+    else:
+        new_data_arrays = old_data_arrays
+        new_unit = old_unit
+    return new_data_arrays, new_unit
+
+def convert_or_undo_photons_per_sec_to_radiance(
+        data_quantities, obs_wavelength, detector_type,
+        spectral_dispersion_per_pixel, solid_angle, undo=False):
+    """
+    Converts data quantities from counts/s to radiance (or vice versa).
+
+    Parameters
+    ----------
+    data_quantities: iterable of `astropy.units.Quantity`s
+        Quantities to be converted.  Must have units of counts/s or
+        radiance equivalent counts, e.g. erg / cm**2 / s / sr / Angstrom.
+
+    obs_wavelength: `astropy.units.Quantity`
+        Wavelength at each element along spectral axis of data quantities.
+
+    detector_type: `str`
+        Detector type: 'FUV', 'NUV', or 'SJI'.
+
+    spectral_dispersion_per_pixel: scalar `astropy.units.Quantity`
+        spectral dispersion (wavelength width) of a pixel.
+
+    solid_angle: scalar `astropy.units.Quantity`
+        Solid angle corresponding to a pixel.
+
+    undo: `bool`
+        If False, converts counts/s to radiance.
+        If True, converts radiance to counts/s.
+        Default=False
+
+    Returns
+    -------
+    new_data_quantities: `list` of `astropy.units.Quantity`s
+        Data quantities converted to radiance or counts/s
+        depending on value of undo kwarg.
+
+    """
+    # Check data quantities are in the right units.
+    if undo is True:
+        for data in data_quantities:
+            assert data.unit.is_equivalent(RADIANCE_UNIT)
```

Comment:
it would be better to raise a proper error with a message rather than just using assert. (Same below)